packages/@aws-cdk/core/test/secret-value.test.ts

```diff
@@ -39,6 +39,21 @@ nodeunitShim({
 
     // THEN
     test.deepEqual(stack.resolve(v), '{{resolve:secretsmanager:secret-id:SecretString:::}}');
+    test.equal( undefined, v.secretArn);
+    test.done();
+  },
+
+  'secretsManager with an ARN ID stores ARN'(test: Test) {
+    // GIVEN
+    const stack = new Stack();
+    const secretId = 'arn:aws:secretsmanager:us-west-2:123456789012:secret:mysecret-123456';
+
+    // WHEN
+    const v = SecretValue.secretsManager(secretId, { jsonField: 'json-key' });
+
+    // THEN
+    test.deepEqual(stack.resolve(v), `{{resolve:secretsmanager:${secretId}:SecretString:json-key::}}`);
+    test.equal(`${secretId}:json-key`, v.secretArn);
     test.done();
   },
```

Comment:
I feel like we're missing tests for `SecretValue` directly obtained from a `Secret` using methods like these: https://github.com/aws/aws-cdk/blob/6179ac16f2fcab5bf5e03a3629db281018d7585b/packages/%40aws-cdk/aws-secretsmanager/lib/secret.ts#L41-L50

Especially the cross-account case.

Reply:
@skinny85 do you have a hint on how to properly integrate the dependency to `secretsmanager` for this test?

Reply:
Sorry, I meant that these tests should be in the CodeBuild module, not in the `@aws-cdk/core` module.